### ISSUE TITLE
Install chef-server-ctl install opscode-manage 12 rc4 behind a proxy AND https transport for ubuntu

### ISSUE THREAD

**Issue by stevendanna** [chef/chef-server#191]
In chef/opscode-omnibus#537 @ehaselwanter said:

> I'm having trouble installing the opscode-manage add-on behind a proxy.

> tried *everything* ;-)

> `/etc/environment`

> ```
http_proxy=http://172.XX:3128
ftp_proxy=http://172.XX:3128
https_proxy=http://172.XX:3128
HTTP_PROXY=172.XX:3128
HTTPS_PROXY=172.XX:3128
```

> ```
pp ENV.to_hash
```

> - right before the remote_file resource gives my proxy settings
- added them to /etv/chef/solo.rb
- added them to /etc/chef/client.rb

> so they are there, but somehow do not work

> ```
Errno::EHOSTUNREACH
-------------------
No route to host - connect(2)

> Resource Declaration:
---------------------
# In /opt/opscode/embedded/cookbooks/apt/providers/repository.rb

> 60:     remote_file cached_keyfile do
 61:       source new_resource.key
 62:       mode 00644
 63:       action :create
 64:     end
 65:   else
```

> I've seen lots of issues around proxies and just wanted to let you know that the embedded
```
chef-server-ctl install opscode-manage
Starting Chef Client, version 11.12.2
```

> does *not* do the right thing.

> https://github.com/opscode/chef/issues/2140

> To fix it I commented out the remote_file section and did:

> ```
wget https://packagecloud.io/gpg.key
mv gpg.key /opt/opscode/embedded/cookbooks/cache/gpg.key
```

> and *yes*  wget/curl pick the env settings and work just fine. after that I got a 

> ```
Mixlib::ShellOut::ShellCommandFailed
------------------------------------
Expected process to exit with [0], but received '100'
---- Begin output of apt-get update ----
STDOUT:
STDERR: E: The method driver /usr/lib/apt/methods/https could not be found.
---- End output of apt-get update ----
Ran apt-get update returned 100

> Resource Declaration:
---------------------
# In /opt/opscode/embedded/cookbooks/apt/recipes/default.rb

>  40: execute 'apt-get-update' do
 41:   command 'apt-get update'
 42:   ignore_failure true
 43:   only_if { apt_installed? }
 44:   not_if { ::File.exists?('/var/lib/apt/periodic/update-success-stamp') }
 45: end
 46:
```

> which ignores the failure BUT fails to update apt for the packages

> Finally

> ```
apt-get install apt-transport-https
apt-get update
chef-server-ctl install opscode-manage
opscode-manage-ctl reconfigure
```

> works fine

**Comment by lamont-granquist** [comment]
looks fixed by chef/chef#3213

**Comment by pathirskumar** [comment]
Hi all
I got resolved my issue, with this solution
Thank you so much.
